src/qml/qmlmixxxcontroller.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <QQuickItem>
+
+#include "controllers/rendering/controllerrenderingengine.h"
+
+namespace mixxx {
+namespace qml {
+
+class QmlMixxxController : public QQuickItem {
+    Q_OBJECT
+    QML_NAMED_ELEMENT(Controller)
+    Q_PROPERTY(QJSValue init READ getInit WRITE setInit
+                    NOTIFY initChanged REQUIRED);
+    Q_PROPERTY(QJSValue shutdown READ getShutdown WRITE setShutdown
+                    NOTIFY shutdownChanged REQUIRED);
+    Q_PROPERTY(QJSValue transformFrame READ getTransform WRITE setTransform
+                    NOTIFY transformChanged REQUIRED);
```

Comment:
do we want to support changing these at runtime? Have you thought about marking them `CONSTANT`? Does this still work without `NOTIFY` signal?
[the Qt docs aren't clear on this](https://doc.qt.io/qt-6/qtqml-cppintegration-exposecppattributes.html#notes-on-use-of-notify-signals)

Reply:
> Have you thought about marking them CONSTANT?

We could - I guess there could be value in swapping these functions.

> Does this still work without NOTIFY signal?

AFAIK, it does - `NOTIFY` is only relevant from the QML property binding, but in our case we wouldn't care as we will use the value help by the property when needed, no reactiveness is required.

Reply:
right, I'm asking because leveraging `CONSTANT` and `MEMBER` would simplify the class drastically.

Reply:
Fair enough - I think making `CONSTANT` is fine . Arguably, if there is the need to swap the function at runtime, this can be done with a bid of logic in those function

Reply:
right, and I don't think there is much of a usecase to re-evaluate other code based on the behavior change, so we can skip the `NOTIFY` signal (which we need to for the `CONSTANT`).